src/bootstrap/bootstrap.py

```diff
@@ -1379,12 +1375,28 @@ def main():
     args = parse_args(sys.argv)
     help_triggered = args.help or len(sys.argv) == 1
 
-    # If the user is asking for help, let them know that the whole download-and-build
+    # Root help (e.g., x.py --help) prints help from the saved file to save the time
+    if help_triggered and sys.argv[1] in ["-h", "--help"]:
+        try:
+            with open(
+                os.path.join(os.path.dirname(__file__), "../etc/xhelp"), "r"
+            ) as f:
+                # The file from bootstrap func already has newline.
+                print(f.read(), end="")
+                sys.exit(0)
+        except Exception as error:
+            eprint(
+                f"ERROR: unable to run help: {error}\n",
+                "x.py run generate-help may solve the problem.",
+            )
+            sys.exit(1)
+
+    # If the user is asking for other helps, let them know that the whole download-and-build
     # process has to happen before anything is printed out.
     if help_triggered:
         eprint(
-            "INFO: Downloading and building bootstrap before processing --help command.\n"
-            "      See src/bootstrap/README.md for help with common commands."
+            "INFO: Downloading and building bootstrap before processing --help command.\n",
+            "See src/bootstrap/README.md for help with common commands.",
```

Comment:
I think the indent/lack of comma here was intentional; why remove it?

Reply:
I modified them unexpectedly. I fixed it in https://github.com/rust-lang/rust/pull/146692/commits/9e446828a8d106ed171ef2e5c0bdcb148a75619b